packages/vaadin-virtual-list/src/virtualizer-iron-list-adapter.js

```diff
@@ -0,0 +1,430 @@
+      // Scrolling by "pages" instead of pixels
+      deltaY *= this._scrollPageHeight;
+    }
+
+    this._deltaYAcc = this._deltaYAcc || 0;
+
+    if (this._wheelAnimationFrame) {
+      // Accumulate wheel delta while a frame is being processed
+      this._deltaYAcc += deltaY;
+      e.preventDefault();
+      return;
+    }
+
+    deltaY += this._deltaYAcc;
+    this._deltaYAcc = 0;
+
+    this._wheelAnimationFrame = true;
+    this.__debouncerWheelAnimationFrame = Debouncer.debounce(
+      this.__debouncerWheelAnimationFrame,
+      animationFrame,
+      () => (this._wheelAnimationFrame = false)
+    );
+
+    const momentum = Math.abs(e.deltaX) + Math.abs(deltaY);
+
+    if (this._canScroll(this.scrollTarget, e.deltaX, deltaY)) {
+      e.preventDefault();
+      this.scrollTarget.scrollTop += deltaY;
+      this.scrollTarget.scrollLeft += e.deltaX;
+
+      this._hasResidualMomentum = true;
+
+      this._ignoreNewWheel = true;
+      this._debouncerIgnoreNewWheel = Debouncer.debounce(
+        this._debouncerIgnoreNewWheel,
+        timeOut.after(this.timeouts.IGNORE_WHEEL),
+        () => (this._ignoreNewWheel = false)
+      );
+    } else if ((this._hasResidualMomentum && momentum <= this._previousMomentum) || this._ignoreNewWheel) {
+      e.preventDefault();
+    } else if (momentum > this._previousMomentum) {
+      this._hasResidualMomentum = false;
+    }
+    this._previousMomentum = momentum;
+  }
+
+  /**
+   * Determines if the element has an ancestor that handles the scroll delta prior to this
+   *
+   * @private
+   */
+  _hasScrolledAncestor(el, deltaX, deltaY) {
+    if (el === this.scrollTarget || el === this.scrollTarget.getRootNode().host) {
+      return false;
+    } else if (
+      this._canScroll(el, deltaX, deltaY) &&
+      ['auto', 'scroll'].indexOf(getComputedStyle(el).overflow) !== -1
+    ) {
+      return true;
+    } else if (el !== this && el.parentElement) {
+      return this._hasScrolledAncestor(el.parentElement, deltaX, deltaY);
+    }
+  }
+
+  _canScroll(el, deltaX, deltaY) {
+    return (
+      (deltaY > 0 && el.scrollTop < el.scrollHeight - el.offsetHeight) ||
+      (deltaY < 0 && el.scrollTop > 0) ||
+      (deltaX > 0 && el.scrollLeft < el.scrollWidth - el.offsetWidth) ||
+      (deltaX < 0 && el.scrollLeft > 0)
+    );
+  }
+
+  /**
+   * @returns {Number|undefined} - The browser's default font-size in pixels
+   * @private
+   */
+  _getScrollLineHeight() {
+    const el = document.createElement('div');
+    el.style.fontSize = 'initial';
+    el.style.display = 'none';
+    document.body.appendChild(el);
+    const fontSize = window.getComputedStyle(el).fontSize;
+    document.body.removeChild(el);
+    return fontSize ? window.parseInt(fontSize) : undefined;
+  }
+
+  __getVisibleElements() {
+    return Array.from(this.elementsContainer.children).filter((element) => !element.hidden);
+  }
+
+  /** @private */
+  __reorderElements() {
```

Comment:
This function reorders the physical elements in the DOM. The primary purpose is to maintain the correct tabbing order and a11y. The logic was moved from [here](https://github.com/vaadin/web-components/blob/b163d3a8c200e6a9064e03b7ca1c4a34ebb4f3d9/packages/vaadin-grid/src/vaadin-grid-scroll-mixin.js#L281)